### ISSUE TITLE
Focus "collision" between form validation and header/* focus on page render

### ISSUE THREAD

**Issue by thiessenp-cds** [cds-snc/platform-forms-client#5263]
An accessibility best practice is to focus the page heading (or first input or..) on load to help a user know where they are. Currently there is a focus "collision" between form validation and header/* focus. So what is focussed is unpredictable.

Since the page has previously loaded, the user knows where they. When a form validation with an error occurs that causes a page re-render, the form validation should be focussed (and not the header).

**Comment by Abi-Nada** [comment]
Once a fix is in, we should also follow up with Freshdesk ticket [22389](https://gcdigital.slack.com/archives/C03T4FAE9FV/p1745947379670959): the focus starts on the first question (and not the page text at the beginning of the page):

https://github.com/user-attachments/assets/d30df250-0b2f-4708-989b-c11375a581c8

**Comment by Abi-Nada** [comment]
Hi @timarney - looks like this issue still persists:

https://github.com/user-attachments/assets/dfe76fa1-4b65-4661-be5b-d2ed569b4972

**Comment by timarney** [comment]
PR 
https://github.com/cds-snc/platform-forms-client/pull/5794

**Comment by Abi-Nada** [comment]
Tested and LTM
